### ISSUE TITLE
MA0001 is not reported when String.GetHashCode() is used

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#528]
### Version of the Meziantou.Analyzer NuGet package

2.0.50

### Rule Identifier

MA0001

### Target Framework

net7.0

### C# Language version

C# 11

### Description

When `String.GetHashCode()` is used without specifying a **StringComparison**, then **MA0074** is reported, I would have expected **MA0001** to be reported since that one is documented as an equivalent of **CA1307** (which is also reported for the same code).

Would it be possible to clarify the rationale of **MA0001** and **MA0074**?
The fact that the doc pages for both rules are identical does not make it any more clearer :p

**Note:**
As mentioned, **CA1307** is currently documented as an equivalent of **MA0001** but it may rather be equivalent of **MA0001** and **MA0074** combined.

### Reproduction Steps

Minimal code:

```c#
var hashCode = string.Empty.GetHashCode();
```

**Comment by meziantou** [comment]
MA0001 and MA0074 are complementary. Initially, everything was reported as MA0001. Then, the rule has been broken up into 2 different rules, so the level could be adjusted.

Both rules report when there is an overload with an additional parameter of type `StringComparison`. The difference is the behavior of the method when the argument is not set:
- MA0001 (info): The default comparison is `Ordinal`
- MA0074 (warn): The default comparison is `CurrentCulture` 

The main goal is to avoid using implicit culture sensitive operations. So, MA0001 is only a suggestion, while MA0074 is a warning.

````c#
"".Equals("");     // MA0001 as the current code use Ordinal implicitly
"".StartsWith(''); // MA0001 as the current code use Ordinal implicitly
"".StartsWith(""); // MA0074 as the current code use CurrentCulture implicitly
````
